test/new-e2e/system-probe/system-probe-test-env.go

```diff
@@ -44,42 +50,55 @@ type TestEnv struct {
 }
 
 var (
-	CustomAMIWorkingDir  = filepath.Join("/", "home", "kernel-version-testing")
-	SSHKeyFile           = filepath.Join(".", "/", "aws-ssh-key")
-	vmConfig             = filepath.Join(".", "system-probe", "config", "vmconfig.json")
+	CustomAMIWorkingDir = filepath.Join("/", "home", "kernel-version-testing")
+	vmConfig            = filepath.Join(".", "system-probe", "config", "vmconfig.json")
+
 	DD_AGENT_TESTING_DIR = os.Getenv("DD_AGENT_TESTING_DIR")
+	CI_PROJECT_DIR       = os.Getenv("CI_PROJECT_DIR")
 	sshKeyX86            = os.Getenv("LibvirtSSHKeyX86")
 	sshKeyArm            = os.Getenv("LibvirtSSHKeyARM")
+
+	stackOutputs = filepath.Join(CI_PROJECT_DIR, "stack.outputs")
 )
 
-func NewTestEnv(name, securityGroups, subnets, x86InstanceType, armInstanceType string, opts *SystemProbeEnvOpts) (*TestEnv, error) {
-	systemProbeTestEnv := &TestEnv{
-		context: context.Background(),
-		name:    fmt.Sprintf("microvm-scenario-%s", name),
+func outputsToFile(output auto.OutputMap) error {
+	f, err := os.Create(stackOutputs)
+	if err != nil {
+		return fmt.Errorf("failed to create file: %s: %w", stackOutputs, err)
 	}
+	defer f.Close()
 
-	sshkey, err := runner.GetProfile().SecretStore().Get("aws_ec2_kitchen_ssh_key")
-	if err != nil {
-		return nil, fmt.Errorf("aws get credential: %w", err)
+	w := bufio.NewWriter(f)
+
+	for key, value := range output {
+		fmt.Fprintf(w, "%s %s\n", key, value.Value.(string))
 	}
+	w.Flush()
 
-	// Write ssh key to file
-	f, err := os.Create(SSHKeyFile)
-	if err != nil {
-		return nil, fmt.Errorf("ssh key create: %w", err)
+	return nil
+}
+
+func GetEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+func NewTestEnv(name, x86InstanceType, armInstanceType string, opts *SystemProbeEnvOpts) (*TestEnv, error) {
+	var err error
+	systemProbeTestEnv := &TestEnv{
+		context: context.Background(),
+		name:    fmt.Sprintf("microvm-scenario-%s", name),
 	}
-	defer f.Close()
-	f.WriteString(sshkey)
 
 	stackManager := infra.GetStackManager()
 
 	config := runner.ConfigMap{
+		"ddinfra:env":                            auto.ConfigValue{Value: opts.InfraEnv},
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
We should discuss about possibly moving this to a `TestSuite` that `go test` could run, vs using `go run` with a custom executable go program.
We are moving towards `TestSuite` to help handling configmap / upresult parsing. 

In the meantime, I suggest using keys from test_infra_definitions - you can find an example is https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/runner/configmap.go#L20-L24

Could you consider using the configmap from test helpers, in a follow up ?

Reply:
Can you elaborate which configmap you are referring to?